### PR TITLE
Feature/hocs 2611 k8s refinement

### PR DESCRIPTION
**Pull request by philbarton**
Sleep in response to SIGTERM  to allow jobs to finish

Swallow exceptions after successful document storage and case creation.

### CODE REVIEW

File: scripts/run.sh [hunk 1]
```diff
@@ -1,6 +1,7 @@
-#!/usr/bin/env bash
+#!/bin/bash
```
Comment (zuzak) [inline scripts/run.sh:1]: This is less portable.

Comment (philbarton) [reply scripts/run.sh]: I've changed it back to what it was.

**Review by zuzak — CHANGES_REQUESTED** [review]
Liveness endpoints should only be used to answer the question "Does this container need restarting?". Using it instead to answer "Can I safely kill this pod?" isn't what it's designed for, and I can imagine it causing subtle problems down the line. It might be faster to boot/kill this way, but we're never in a rush in production; I'd rather a simple solution that took a couple more seconds.

If you want to ensure your container cleans things up before termination, listen for `SIGTERM` and perform your operations there. If you don't think your application can do that within 30 seconds, adjust the `terminationGracePeriodSeconds` at the pod spec level.

Alternatively, use a [`preStop`](https://kubernetes.io/docs/concepts/containers/container-lifecycle-hooks/#hook-details) hook to define a command to run that clears everything up.

I don't like the idea of merely checking the presence of a file for our liveness check. The successful ability of `cat` to run has no bearing on whether the application itself is unhealthy.  A better idea might be to write the system time to a file, and check periodically that that time is up to date.

**Comment by philbarton** [comment]
Thanks for the clarification. I've removed the health file and the probes. As this is a message consumer, liveness can only be measured by something like a high queue depth. Maybe too much for now.

Spring behaviour on SIGTERM is to exit. I've added a sleep to let threads finish.

The improved startup and shutdown I saw is more likely related to doing an 'exec java'. This makes spring run as PID 1 which means k8s has a more accurate picture of the apps state, not just the outer shell. We should probably do that for all apps.

**Review by robertdeniszczyc2 — APPROVED** [review]
Looks good to me

File: src/main/java/uk/gov/digital/ho/hocs/config/AppCustomProperties.java [hunk 2]
```diff
@@ -19,6 +19,8 @@
     @NotBlank
     private String basicAuth;
     @NotBlank
+    private Integer shutdownDelaySeconds;
```
[the remark this answers is not in the document]
Comment (philbarton) [reply src/main/java/uk/gov/digital/ho/hocs/config/AppCustomProperties.java:22]: it is, it still sleeps, because without it spring dies on SIGTERM

**Pull request merged**